[13.0][MIG] product_pack

jjscarafia opened the pull request.

Commit:
INIT

INIT

ADD pot files and module for agrimesores

Commit:
[FIX] traslation and clean modules

jjscarafia committed:
[IMP/FIX] product pack

IMP product pack to support surveyor packs

IMP pack description

IMP add product pack standar functionality for v8 and general improvements

IMP discounts and portal distributors permissions

FIX product_pack license and author

FIX prod pack available

IMP pack to restrict company in products

FIX pack company restriction

Commit:
IMP

jjscarafia committed:
[IMP/FIX] product_pack

IMP product_pack format

FIX fix_digits_compute_to_digits

IMP IMPORTANT imprrovements on product pack, get pack with sale order constraint

IMP use constraint on sale line and not sale order for product packs

IMP implement totalice price on product pack

IMP update pack manifest

FIX prod pack copy sale order

FIX contraint

IMP compt with other modules

FIX prod pack @api.multi

adhoc-transbot committed:
OCA Transbot updated translations from Transifex

nicomacr committed:
9.0-fix-product_pack (#110)

9.0 FIX filter the taxes by company (#111)

jjscarafia committed:
FIX product pack on sale orders with pop up for lines

adhoc-transbot committed:
[MIG] Make modules uninstallable

jjscarafia committed:
[REL] init repository v11 (#122)

* [REL] replace __openerp__ for __manifest__

* [REL] replace openerp imports for odoo

* [REL] some python3 exceptions

* [REL] replace openerp by odoo on xml files

* [REL] delete utf-8 line on python files

* [REL] some minor fixes for migration to v11

* [FIX] runbot link

nicomacr committed:
[REL] migration product_pack to 11.0  (#145)

* [FIX] xml files names

* [REL] migration product_pack to 11.0
* [FIX] Update README
* [FIX] remove openerp tags and data tags
* [IMP] change some funtion to improve funcionality
* [FIX] change name to xml and .py files according guidelines
* [FIX] remove api.one decorators

* [FIX] update readme

* [FIX] Remove "encoding: utf-8"

* [FIX] remove api.multi, string not necesary and some other fixes

* [FIX] demo data to separate according Guidelines

* [FIX] fix names in fields

* [FIX] remove executable

jjscarafia committed:
[IMP] product pack access rights

nicomacr committed:
[FIX] compatibility with price_security and product_pack with discount in lines

zaoral committed:
[ADD] product_pack: Compatibility 11.0 and multiples improves. (#185)

* [ADD] product_pack: Add compatibility with website_sale.

* [ADD] product_pack: Allow product pack modify

* [ADD] product_pack: In sale order do not let to delete/modify a line that
belongs to a pack

* [ADD] product_pack: Be able to edit pack lines inf allow_modify_pack is set in
product.

* [ADD] product_pack: Do not compute price in cart for components_price packs

[FIX] product_pack: Pack line price fixed (take into account pricelist) (#187)

* [FIX] product_pack: Pack line price fixed (take into account pricelist)

Proper compute pack line price using price instead of list_price

* [FIX] product_pack: Allow Modidy Pack only showed for Components price packs

* [FIX] product_pack: When changing pack_price_type clean allow_modify_pack if
needed.

nicomacr committed:
[FIX] the stock calculation from product packs if see from product template view (#189)

[FIX] we add prefetch_fields becase if not the stock it's no correct when this method it's call from website (#191)

zaoral committed:
[FIX] product_pack: Take into account both pricelist and pack line discount. (#197)

Formerly was only taking into account the product packs discount, now take
into account both discounts.

Also we simplify the way that get_sale_order_line_vals works and now is
simplier and use Odoo's onchange methods to compute the lines values.

[FIX] product_pack: Pack component now have the tax of the company (#198)

Before this fix the product's packs new sale order line have multiple taxes:
one por each company. Actually only need to add the taxes realated to the
sale order's company.

[FIX] product_pack: compute proper components prices (#199)

In some cases the values were prefetched and it was throwing bad prices.
we add prefetch_fields to the context of the packs and that solve it.

[FIX/IMP] product_pack: Allow modify pack and do not reset after confirm website sale order (#201)

* [FIX] product_pack: Do not reset packs components when confirm sale order.

If a sale order is updated from the website whrn confirm the order we do
not expand the pack, we usit as it is.

* [ADD] product_pack: Allow modify pack from backend/website

Now we can choose if the product can be modified in the only int he backend or
if can be modified also in the frontend by the customers.

* [FIX] lint

jjscarafia committed:
[IMP] product pack compatibility with ecommerce and update prices (#202)

nicomacr committed:
[FIX] product_pack: remove price and discount if the parent pack it's the type 'fixed and totalice' (#203)

ernestotejeda committed:
[ADD] product_pack: allows you to set product packs

ernestotejeda committed:
[ADD] stock_product_pack: new module

oca-travis committed:
[UPD] Update product_pack.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

ernestotejeda committed:
[IMP] product_pack: convert amount to price list currency

zeroheure committed:
[IMP] product_pack: better english, fix error in steps (#5)

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

oca-travis committed:
[UPD] Update stock_product_pack.pot

ernestotejeda committed:
[IMP] stock_product_pack: improve description and usage readme section

pedrobaeza committed:
[MIG][FIX] product_pack: Extra comma in SQL

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 95.6% (43 of 45 strings)

Translation: product-pack-12.0/product-pack-12.0-product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-12-0/product-pack-12-0-product_pack/pt/

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-pack-12.0/product-pack-12.0-stock_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-12-0/product-pack-12-0-stock_product_pack/pt/

pedrobaeza committed:
[MIG][FIX] product_pack: Avoid to put values on non packs

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (45 of 45 strings)

Translation: product-pack-12.0/product-pack-12.0-product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-12-0/product-pack-12-0-product_pack/ca/

jjscarafia committed:
[IMP] product-pack: black, isort, prettier

jjscarafia committed:
[MIG] product_pack/: Migration to 13.0

pedrobaeza commented:
Please see Travis status.

jjscarafia commented:
Hi @pedrobaeza thanks for the review. Can you help me a little bit? It seams to bre related to new pre-commit implementation but I didn't found documentation regarding this.
1. I've already run the "pre-commit run -a" as pointed [here](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch)
2.Some lint errors I can fix and push again, but I don't understand [this one](https://travis-ci.com/github/OCA/product-pack/jobs/317894083#L321)


Thanks!

jjscarafia commented:
fixed @pedrobaeza 
thanks!

nicomacr committed:
[12.0] [FIX] product_pack : use sudo in the get_pack_lines method

This is because in the e-commerce when an user portal or public try to access to the catalog of products has an access error to the model "product pack line".
 Them problem is when the method price_compute is call then try to compute the products expand the product pack and try to access to that model.
This way evoid to set an access  read to "product pack line" for the portal and public users.

ernestotejeda commented on product_pack/models/product_pack_line.py:
```diff
@@ -0,0 +1,51 @@
+            while pack_lines:
+                if parent_product in pack_lines.mapped("product_id"):
+                    raise ValidationError(
+                        _("You cannot set recursive packs.\n" "Product id: %s")
```
```suggestion
                        _("You cannot set recursive packs.\nProduct id: %s")
```

jjscarafia replied:
done @ernestotejeda thanks!

ernestotejeda commented on product_pack/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    "category": "Product",
+    "summary": "This module allows you to set a product as a Pack",
+    "website": "https://github.com/OCA/product-pack",
+    "author": "NaN·tic, " "ADHOC SA, " "Tecnativa, " "Odoo Community Association (OCA)",
```
```suggestion
    "author": "NaN·tic, ADHOC SA, Tecnativa, Odoo Community Association (OCA)",
```
I know that black does things like this but they look a bit ugly and if we fixes it like that black won't change it again

ernestotejeda reviewed (changes requested):
Thanks @jjscarafia :+1: 

- Remove product_pack/migrations/12.0.1.0.0 folder

pedrobaeza commented:
Please include #15

azucenatrey committed:
[FIX] stock_product_pack: only use stockable products to calculate the stock.

pedrobaeza commented:
Please include as well #18

OCA-git-bot committed:
stock_product_pack 12.0.1.0.1

ypapouin committed:
Added translation using Weblate (French)

ypapouin committed:
Translated using Weblate (French)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-pack-12.0/product-pack-12.0-stock_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-12-0/product-pack-12-0-stock_product_pack/fr/

jjscarafia commented:
> Thanks @jjscarafia 
> 
> * Remove product_pack/migrations/12.0.1.0.0 folder

I're removed them but I don't get the point of doing it. We use odoo upgrade service and we often jump versions, for eg. we send odoo a v11 db and we get a v13 one, and we would like migration scriptf for v12 to be run while upgrading to v13

Any opinion regarding this @ernestotejeda @pedrobaeza ?

jjscarafia committed:
[FIX] product_pack: black prettier

jjscarafia commented:
all the changes done. Thanks @pedrobaeza @ernestotejeda 

@pedrobaeza https://github.com/OCA/product-pack/pull/18/files will be considered on new pr for stock_product_pack

jjscarafia committed:
[IMP] stock_product_pack: black, isort, prettier

jjscarafia committed:
[MIG] stock_product_pack: Migration to 13.0

jjscarafia commented:
I've closed this PR and open a new one with same content because travis was red and I couldn't achive to re run it without a new PR.
https://github.com/OCA/product-pack/pull/22
